django_freeradius/base/admin.py

```diff
@@ -59,10 +61,32 @@ class AbstractRadiusGroupUsersAdmin(TimeStampedEditableAdmin):
 
 
 class AbstractRadiusCheckAdmin(TimeStampedEditableAdmin):
-    list_display = ('username', 'attribute', 'value', 'is_active',
-                    'created', 'modified')
-    search_fields = ('username',)
-    list_filter = ('created', 'modified')
+    list_display = (
+                    'username', 'attribute', 'is_active',
+                    'created', 'valid_until',
+                   )
+    search_fields = ('username', 'value')
+    list_filter = (
+                    DuplicateListFilter, ExpiredListFilter, 'created',
+                    'modified', 'valid_until'
+                    )
+    readonly_fields = ('value',)
+    form = AbstractRadiusCheckAdminForm
+    fields = ('username', 'value', 'op', 'attribute', 'new_value',
+              'is_active', 'valid_until', 'note', 'created', 'modified')
+
+    # if new_value is present this will be hashed and stored into value
+    def save_model(self, request, obj, form, change):
+        password_renewed = form.data['new_value']
```

Comment:
why not put this logic in the model so it's easier to test and reuse across different views?

Reply:
Yes, I would like to suggest an PasswordEncoderHelper, out of there :)